app/src/main/java/com/hedvig/app/util/extensions/ContextExtensions.kt

```diff
@@ -285,3 +288,23 @@ fun Context.tryOpenUri(uri: Uri) {
         showError()
     }
 }
+
+fun Context.startChat(
+    trackingFacade: TrackingFacade,
+    closable: Boolean = true,
+    restartable: Boolean = false
+) {
+    val intent = Intent(this, ChatActivity::class.java)
+    intent.putExtra(ChatActivity.EXTRA_SHOW_CLOSE, closable)
+    intent.putExtra(ChatActivity.EXTRA_SHOW_RESTART, restartable)
+
+    val options =
+        ActivityOptionsCompat.makeCustomAnimation(
+            this,
+            R.anim.chat_slide_up_in,
+            R.anim.stay_in_place
+        )
+
+    ActivityCompat.startActivity(this, intent, options.toBundle())
+    trackingFacade.track("click_start_chat")
+}
```

Comment:
Hmm this now makes it so that wherever we need to open the chat, we need to have a hold of the trackingFacade which might be a bit awkward in some contexts (let's say inside a composable inside a ViewHolder).
Would it make sense that we instead log this event inside ChatActivity instead, where it can fetch its own instance of the tracker? If not, is it a problem with duplicate events or something similar?

Reply:
You're right, I'll change! Good input :)